### ISSUE TITLE
Give page the right lang attribute

### ISSUE THREAD

**Issue by upsuper** [navidrome/navidrome#2174]
### Is your feature request related to a problem? Please describe.

Currently, the page always have `<html lang="en">` regardless of what language the UI is using.

### Describe the solution you'd like

Each language in [resources/i18n](https://github.com/navidrome/navidrome/tree/2400e4f60da705888298cb493948b1c008a1980c/resources/i18n) should include an [IETF BCP 47 language tag](https://en.wikipedia.org/wiki/IETF_language_tag), for example, Japanese is `ja`, Simplified Chinese is generally `zh-CN` and Traditional Chinese is `zh-TW`.

When the UI language is updated, the `lang` attribute should be updated to the corresponding language tag.

### Describe alternative solutions that would also satisfy this problem

Not aware of.

### Additional context

This is a problem because language attribute affects
* a11y, e.g. screen reader,
* fallback font choice,
  * which is especially a problem for CJK languages as they share Chinese characters, but different fonts have different character coverage, and the same character can come with different shape in different languages,
* text transform, e.g. uppercasing,
  * which is especially a problem for Latin-based languages, because different languages have different transformation rules, for example,
    * in Turkish, the uppercase of "i" is "İ" rather than "I"
    * in Irish, the uppercase of "gc" is "gC" not "GC"

(Though my main concern is on CJK.)

**Comment by deluan** [comment]
Yes, I think this can be done dynamically, as the `<html lang="en">` tag comes from the static template in the server and the language is a property of the UI javascript client. 

For reference: https://stackoverflow.com/a/52287911

**Comment by subhajit20** [comment]
@deluan hey! can you tell me that is there any option in the application through which we can update the language of the software ? or the language will be updated organically ?

**Comment by deluan** [comment]
@subhajit20 Sorry, what language are you referring to? The `lang` property of the HTML as discussed above or the translation?

**Comment by subhajit20** [comment]
Yes ! Do I have to populate all the language inside the lang attribute or anything else I am pretty confused 😕

**Comment by deluan** [comment]
Sorry again, but I'm not sure if you are trying to fix this issue or asking support. 

Assuming you want to fix the issue with a PR, you'd have to get the current language from [`locale` key in localStorage](https://github.com/navidrome/navidrome/blob/310c816cdd21b568fe51ce18d0465d6dac84c7f3/ui/src/i18n/provider.js#L9) and change the `html lang` attribute dynamically as per the SO link I posted above.  This would probably have to happen in a `useEffect` call back in the main [App.js](https://github.com/navidrome/navidrome/blob/310c816cdd21b568fe51ce18d0465d6dac84c7f3/ui/src/App.js#L145) *AND* when the user selects a new language in [`SelectLanguage.js` file](https://github.com/navidrome/navidrome/blob/310c816cdd21b568fe51ce18d0465d6dac84c7f3/ui/src/personal/SelectLanguage.js#L33).

This of course assuming that changing the property dynamically actually solves the issues mentioned by @upsuper . If it does not, and we have to actually change the `index.html` template, the solution will be way more complex.

**Comment by subhajit20** [comment]
@deluan  yes this is what I wanted to know and yeas I am intend to solve the issue that's why I was asking.
1. If I get the language key from the local storage of the browser, i could add that keu inside the lang attribute right ?
2. Even if the user change the language then it will also reflect the lang key word and change that into a perticular language as user selects that right ?

**Comment by deluan** [comment]
1. Yes, that's how it should be implemented.
2. When the user selects a new language, the new value is stored in the localStorage (see the link to `SelectLanguage.js` above. In this case you would also need to update the lang attribute.

**Comment by subhajit20** [comment]
@deluan  okay now I get to know the entire scenario, thanks for making it understandable to me .
I will try to fix this issue meanwhile if I need any help please keep in touch with me.

**Comment by subhajit20** [comment]
@deluan hey is it okay ? 
![SelectLanguage js - navidrome - Visual Studio Code 05-04-2023 11_15_03](https://user-images.githubusercontent.com/99127578/229991888-22e00d1b-6a54-444f-94e0-92e19a98bd5c.png)

**Comment by subhajit20** [comment]
@deluan and see it also
![SelectLanguage js - navidrome - Visual Studio Code 05-04-2023 11_22_44](https://user-images.githubusercontent.com/99127578/229992844-b3d1cc96-ef40-46c5-9c89-eb3ca6c8638e.png)

**Comment by deluan** [comment]
Thanks for taking a look at this! The change in `SelectLanguage.js`  is correct, but the one in the `App.js` is wrong, because it would only be called _if_ the ActivityPanel is enabled, which may not always be the case, it is unrelated to the language selection. It would be better to put this in line 145 (the one I linked before), in the beginning of the `AppWithHotkeys` component. That's the entry point of the whole UI application.

As a side not, please do not post screenshots of code. The proper way to share it is using markdown code blocks. See https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code

**Comment by subhajit20** [comment]
okay sure I will fix that and sorry for uploading images as I am new to the world of open source.

**Comment by subhajit20** [comment]
@deluan  another thing is I have read the contribution guide line and I am little confused that should I create a branch name starts with the entire title/issue-number like this ? or anything else ?
Give page the right lang attribute/2174

**Comment by subhajit20** [comment]
@deluan  in the line number 145 there is a if statement then do I have to put that line inside the if statement of before the if statement?

```
const AppWithHotkeys = () => {
  if (config.enableSharing && shareInfo) {
    let language = localStorage.getItem("locale");
    document.documentElement.lang = language;
    return <SharePlayer />
  }
  return (
    <HotKeys keyMap={keyMap}>
      <App />
    </HotKeys>
  )
}
```

**Comment by deluan** [comment]
> I create a branch name starts with the entire title/issue-number like this ? or anything else ?

Check other PRs. There's no hard rule here, but you have to reference this issue, maybe just putting a #2174 in the PR's description.

>  in the line number 145 there is a if statement then do I have to put that line inside the if statement of before the if statement?

If you put it inside the if, it will only work _if_ sharing is enabled, which does not make sense, right?

**Comment by subhajit20** [comment]
@deluan hey whenever I am going to push the commit it shows this error right below

```

E:\Open Source Contribution\navidrome>git push origin Give_page_the_right_lang_attribute/2174
.git/hooks/pre-push: line 4: make: command not found
error: failed to push some refs to 'https://github.com/subhajit20/navidrome.git'
```

**Comment by subhajit20** [comment]
@deluan  hey I have made another fresh pr request.
Can you check that once

**Comment by deluan** [comment]
Closed by #2299. @upsuper Can you confirm this works as intended?

**Comment by upsuper** [comment]
I checked Japanese and two Chinese ones, and I can confirm the font stack works as expected, so I would say it works.

(While testing, though, I noticed that Arabic doesn't switch the page to RTL, but this is probably a separate issue.)
